### PR TITLE
Move out TF and replace FastAPI with Flask

### PR DESCRIPTION
**Pull request by rit1217**
Remove TensorFlow and Keras from data preprocessing. 
Use Flask instead of FastAPI.
Add services and utils files for React web.

### CODE REVIEW

File: api/api.py [hunk 1]
```diff
@@ -1,39 +1,29 @@
-from fastapi import FastAPI
-from pydantic import BaseModel
+import flask
+import flask_cors
```
Comment (IDDT) [inline api/api.py:2]: No need for cors stuff if flask is serving the html template that links the js bundle from flask static dir.

File: api/api.py [hunk 2]
```diff
@@ -1,39 +1,29 @@
...
-@app.post("/complete/")
-def suggest_complete(prefix: InputModel):
-    r = t.autocomplete(prefix.input_str)
-    return r[:20]
\ No newline at end of file
+
+app = flask.Flask(__name__)
+flask_cors.CORS(app)
+
+api_v1 = flask.Blueprint('api_v1', __name__, url_prefix='/api/v1')
```
Comment (IDDT) [inline api/api.py:9]: Let's keep it simple, blueprints and api versions are not needed for this project.

File: api/api.py [hunk 3]
```diff
@@ -1,39 +1,29 @@
...
+
+data = trie_model.data_preprocess()
+
+trie = trie_model.Trie()
+trie.add_data(data)
+
```
Comment (IDDT) [inline api/api.py:15]: Add route to render you html template with js bundle https://flask.palletsprojects.com/en/2.1.x/quickstart/#rendering-templates

File: api/api.py [hunk 4]
```diff
@@ -1,39 +1,29 @@
...
+@api_v1.route('/nextword/', methods = ['POST'])
+def suggest_next_word():
+    req_body = flask.request.get_json()
+    res = trie.get_next_char(req_body['prefix'])[:20]
+    return flask.jsonify(res[:20])
```
Comment (IDDT) [inline api/api.py:20]: I assume it's unused as of now - let's get rid of it.

File: api/api.py [hunk 5]
```diff
@@ -1,39 +1,29 @@
...
+  
+
+@api_v1.route('/complete/', methods = ['POST'])
```
Comment (IDDT) [inline api/api.py:23]:
```suggestion
@app.route('/api/autocomplete/', methods = ['POST'])
```

File: api/api.py [hunk 6]
```diff
@@ -1,39 +1,29 @@
...
+def suggest_complete():
+    req_body = flask.request.get_json()
+    res = trie.autocomplete(req_body['prefix'])
+    return flask.jsonify(res[:20])
```
Comment (IDDT) [inline api/api.py:27]: Pretty sure you can limit it to 20 in your autocomplete function, rather than generating everything and the cutting it here.

**Review by IDDT — CHANGES_REQUESTED** [review]

File: api/api.py (hunk 3 above)
Comment (rit1217) [reply api/api.py:15]: Do you mean to wrapped these lines together with reder_template() with a route, to initialize model?

Comment (rit1217) [reply api/api.py:15]: Oh, I understand now, I will add route for rendering html template and js bundle, not these 3 lines.

**Review by IDDT — APPROVED** [review]

**Pull request merged**